src/btree/bt_cursor.c

```diff
@@ -169,8 +169,7 @@ __cursor_page_pinned(WT_CURSOR_BTREE *cbt, bool search_operation)
      * whether we correctly resolved the transaction becomes hard. It is easier to skip this check
      * in that instance.
      */
-    if (__wt_atomic_load64(&cbt->ref->page->read_gen) == WT_READGEN_OLDEST &&
-      !F_ISSET(session->txn, WT_TXN_PREPARE))
+    if (__wt_evict_page_is_soon(cbt->ref->page) && !F_ISSET(session->txn, WT_TXN_PREPARE))
```

Comment:
This is a good question. We have a lot of eviction terminology in the code that causes confusion. 

Here’s my take:
1. Urgent eviction refers to anything in the urgent queue. I wouldn’t want to rename OLDEST to URGENT since they represent different concepts.

2. During forced eviction - 
    - Place a page in the urgent queue, making it part of **urgent eviction**, or
    - Directly evict the page, making it a **forced eviction**.
    - Forced eviction can only occur if either `READGEN_OLDEST` or `READGEN_WONT_NEED` is set on the page.

Therefore, the logic is:
forced eviction → (`OLDEST` | `WONT_NEED`).

We have renamed `OLDEST` to `EVICT_SOON`, as that better describes its purpose. In terms of priority, "evict soon" feels more immediate than "won't need".

Basically - 

- EVICT_SOON | WONT_NEED → forced eviction.
- EVICT_SOON has an individual getter and setter, which is useful in the code.
- There is a WT_EVICT_CACHE_URGENT flag in forced eviction. Ideally, it should be renamed to WT_EVICT_CACHE_URGENT, but I won’t be addressing this in the current PR.

Reply:
I have removed the word immediate as part of [5464ef4](https://github.com/wiredtiger/wiredtiger/pull/11087/commits/5464ef407b545435ba0c5e78833c4cbf01e356bc)

Reply:
SGTM. You have a better handle on the evict code than I do :).
[These lines](https://github.com/wiredtiger/wiredtiger/pull/11087/files#diff-fba173dbad7997f847117319e1939715f1969181ab2937508a2c9303b201f017R2152) are throwing me off as I assumed `WT_READGEN_OLDEST`/`WT_READGEN_EVICT_SOON` meant "stick this page in the urgent queue ASAP". Perhaps we should also detect `WONT_NEED` pages here but that's outside the PR scope. Resolving the convo

Reply:
I see. I agree, we can handle this aspect later.